apps/web/src/views/Farms/hooks/useLmPoolLiquidity.ts

```diff
@@ -0,0 +1,37 @@
+import { pancakeV3PoolABI } from '@pancakeswap/v3-sdk'
+import { useQuery } from '@tanstack/react-query'
+import { publicClient } from 'utils/wagmi'
+import { Address, parseAbiItem } from 'viem'
+
+const fetchLmPoolLiquidity = async (lpAddress: Address, chainId: number): Promise<bigint> => {
+  const client = publicClient({ chainId })
+  if (!client) {
+    return 0n
+  }
+  try {
+    const lmPool = await client.readContract({
+      address: lpAddress,
+      abi: pancakeV3PoolABI,
+      functionName: 'lmPool',
+    })
+    const lmPoolLiquidity = await client.readContract({
+      address: lmPool,
+      abi: [parseAbiItem('function lmLiquidity() view returns (uint128)')],
```

Comment:
is this the best practice? i don't have much experience, so, just a question

Reply:
it depends, we do not have the `lmPool` ABI defines, it's ok to use only one of them


for more about ABI related api, check the doc start from https://viem.sh/docs/abi/decodeAbiParameters